app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/AccessFactorSourceProxyContract.kt

```diff
@@ -30,10 +36,25 @@ sealed interface AccessFactorSourcesInput {
         val factorSource: FactorSource.CreatingEntity? = null
     ) : AccessFactorSourcesInput
 
-    // just for demonstration - will change in next PR
-    data class ToSign(
-        val someData: List<Int>
-    ) : AccessFactorSourcesInput
+    sealed interface ToReDeriveAccounts : AccessFactorSourcesInput {
+
+        val factorSource: FactorSource.CreatingEntity
+        val isForLegacyOlympia: Boolean
+        val nextDerivationPathOffset: Int // is used as pointer when user clicks "scan the next 50"
+
+        data class WithGivenMnemonic(
+            override val factorSource: FactorSource.CreatingEntity,
```

Comment:
one thing that I want to point out here is the need of consolidating factor sources (or with other objects) which do the same thing. As you can see, in order to create or to recover an account you always need a derivation path and a public key no matter what is the factor source. In this case `CreatingEntity` covers both device and ledger factor sources. 
We do not need to have separate arguments or methods to create an account based on the specific type of factor source. 

And later when we need to derive the public key with a given  `CreatingEntity` we get this 
```
when (factorSource) {
     is Device  -> { ... }
     is Ledger -> { ... }
     // no other unrelated to create account factor sources!
}
```